src/spdx2/agent/spdx2utils.php

```diff
@@ -39,44 +41,58 @@ static public function preWorkOnArgsFlp($args,$key1,$key2)
     return $args;
   }
 
+  static public function getPossiblePrefix($license, $dbManager, $licenseDao, $groupId)
+  {
+    return self::getPossiblePrefixByID($license->getLicenseId(), $dbManager, $licenseDao, $groupId);
+  }
+
+  static public function getPossiblePrefixByID($licenseId, $dbManager, $licenseDao, $groupId)
+  {
+    return $dbManager->booleanFromDb($licenseDao->getLicenseById($licenseId, $groupId)->getSpdxCompatible());
+  }
+
   /**
    * @param string[] $licenses
-   * @param string="" $prefix
+   * @param string[] $prefixMap
    *
    * @return string
    */
-  static public function implodeLicenses($licenses, $prefix="")
+  static public function implodeLicenses($licenses, $prefixMap = null)
   {
     if(!$licenses || !is_array($licenses) || sizeof($licenses) == 0)
     {
       return "";
     }
 
-    $licenses = array_map(function ($license) use ($prefix)
+    $licenses = array_map(function ($license) use ($prefixMap)
     {
       if(strpos($license, " OR ") !== false)
       {
         return "(" . $license . ")";
       }
       else
       {
-        if(is_array($prefix)){
-          $prefix = $prefix[$license];
-        }
-        if(substr($license, 0, strlen($prefix)) === $prefix)
+        if(is_array($prefixMap) &&
+            array_key_exists($license,$prefixMap) &&
+            $prefixMap[$license])
         {
-          return $license;
+          return self::$prefix . $license;
         }
```

Comment:
may be $this instead of self would solve the issue here ??

which is

PHP Fatal error:  Cannot access self:: when no class scope is active in /home/travis/build/siemens/fossology/src/spdx2/agent/spdx2utils.php on line 79